src/include/compat.h

```diff
@@ -72,11 +67,18 @@
 
 #endif /* !__FreeBSD__ */
 
-#if defined(__APPLE__)
-/* PATH_MAX */
+#if defined(__APPLE__) || defined(__FreeBSD__)
+/* get PATH_MAX */
 #include <limits.h>
+
+#ifndef EREMOTEIO
 #define EREMOTEIO 121
+#endif
+
+#ifndef HOST_NAME_MAX
 #define HOST_NAME_MAX 255
+#endif
```

Comment:
and could move the `#endif /* __APPLE__ */` before `#ifndef EREMOTEIO` and change it to `#endif`.